lib/elixir/lib/process.ex

```diff
@@ -421,6 +421,12 @@ defmodule Process do
   @spec register(pid | port, atom) :: true
   def register(pid, name) when not name in [nil, false, true] and is_atom(name) do
     :erlang.register(name, pid)
+  catch
```

Comment:
I think we will need `(is_pid(pid) or is_port(pid))` in the guard to prevent this argument error when not a pid `register(:foo, :foo)`. I think we will also need to exclude `:undefined` because this name will always raise.